### ISSUE TITLE
High CPU usage on macOS

### ISSUE THREAD

**Issue by ajitam** [mcollina/mosca#566]
Hello,

I've been using Mosca for some time know on different ENVs (macOS and Ubuntu). Last week I formatted my Macbook Pro and now whenever I ran nodejs script with Mosca inside `node` task in Activity monitor start burning 50% and more of CPU.

Is there any why I can somehow debug this?

Here is what I know:

* macOS 10.12.1
* node version 6.9.1 (installed over NVM), I also tried with 4.6.0 bd 5.12.0
* I tried restart the computer
* for testing I have empty folder where I install Mosca (`npm install mosca`) and `index.js` with a simple script from the readme file ☛ [index.js](https://github.com/mcollina/mosca#embedded)
* I notices that when I install mosca I get some errors around instaling `zmq`

thx, matija

**Comment by ajitam** [comment]
Ok, I might have found some more leads around this problem.

* I ran script on another mac (node version 5.4.1) and I notice that on **both** systems - if I turn OFF mongoDB backend script runs 0K
* some older scripts on (nodejs 5.4.1) mac were running 0K too but I didn't know why

After some additional research I noticed that - if I ran the script with DB which already existed everything was 0K, but if I started script with new DB CPU usage got hight again. So naturally on newly formatted mac I didn't had any old DBs.

So bottom line: If I start mosca with backend using mongoDB (and use new DB name) CPU usage is hight (DB is created)

**Comment by mcollina** [comment]
@ajitam interesting. Is the CPU staying high, or does it lower after a while?
Maybe there is a bug somewhere in the collection-creation logic. Would you like to try a PR? The problem is likely to be in https://github.com/mcollina/ascoltatori

**Comment by ajitam** [comment]
1. As far as I can tell script is running on high CPU non-stop (I think I was running it for at least 2 hours)

2. I can confirm the same behaviour on Ubuntu 16.04.1 LTS

3. Regarding PR - frankly I'm not filling qualified for the task :/

p.s.: for now I have very dirty fix - I export `mqtt` DB from the server and import it into my local mongoDB
